### PR TITLE
emitting an event when status of timer changes

### PR DESCRIPTION
**Pull request by jbbpatel94**

### CODE REVIEW

**Comment by jbbpatel94** [comment]
**emitting an event when status of timer changes**

this might helpful when we want to do something when timer status changes.

File: src/index.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import EventEmitter from 'events'
 
+const status = {
+  _stopped: 'stopped',
+  _running: 'running',
+  _paused: 'paused'
+}
```
Comment (mathiasvr) [inline src/index.js]: I think it's a good idea to group possible status values,
but I would prefer to remove this for now.

I might do this with Typescript at some point.

File: src/index.js [hunk 2]
```diff
@@ -20,40 +26,45 @@ class Timer extends EventEmitter {
...
   resume () {
-    if (this._status !== 'paused') return
+    if (this._status !== status._paused) return
     this._endTime += Date.now() - this._pauseTime
     this._pauseTime = 0
-    this._status = 'running'
+    this.changeStatus(status._running)
+  }
+  
+  changeStatus (status) {
```
Comment (mathiasvr) [inline src/index.js]: Can we add an underscore, to signal that this is an internal function:
```
  _changeStatus (status) {
```

File: src/index.js [hunk 3]
```diff
@@ -20,40 +26,45 @@ class Timer extends EventEmitter {
...
+    this._status = status
+    this.emit('onChangeStatus', this._status)
```
Comment (mathiasvr) [inline src/index.js]: I would like to rename the event, so it is attached by `timer.on('statusChanged', ...)`
```suggestion
    this.emit('statusChanged', this._status)
```

**Review by mathiasvr — CHANGES_REQUESTED** [review]
Thanks, I like this idea 👍 

I have a couple of change requests that I hope you are good with.

File: src/index.js (hunk 3 above)
Comment (jbbpatel94) [reply src/index.js]: I updated as per your suggestions.

Thanks!

File: src/index.js (hunk 2 above)
Comment (jbbpatel94) [reply src/index.js]: I updated as per your suggestions.

Thanks!

File: src/index.js (hunk 1 above)
Comment (jbbpatel94) [reply src/index.js]: I updated as per your suggestions.

Thanks!

**Comment by jbbpatel94** [comment]
I updated as per your suggestions.

Thanks!

**Comment by mathiasvr** [comment]
@jbbpatel94 Thanks for the quick fixes 👍 

Released it as `v1.3.0`.

**Pull request merged**